addons/mail/static/src/new/activity/activity_service.js

```diff
@@ -53,70 +54,10 @@ export class ActivityService {
      */
     insert(data) {
         const activity = this.store.activities[data.id] ?? new Activity(this.store, data);
-        this.update(activity, data);
+        assignDefined(activity, data);
```

Comment:
Note that in previous code it was not possible to re-assign `id` by misake, whereas it is possible again now.
Similar issues for all fields that were previously considered as readonly, as well as fields that didn't match any key.

For data integrity I much preferred the `update` method from model manager in master.

Reply:
Thankfully all of that is scoped to `insert` functions, which are the entry point of inserting data objects, including managing store from insertion process.

It's allowed just because the code of insert is still rudimentary. I feel it's not a big deal to make it more complex than it should right now, especially if this means far more LOCs.

Reply:
Also, data classes should have improved declaration of fields, to not allow assigning them and/or putting options on the fields, such as readonly. That way, this problem can be fixed in a neat way, without significantly increasing complexity from more LOCs in business code)